File: qwen3.6:27b/qwen3_6_27b/cors_002_naive.py
from flask import Flask

app = Flask(__name__)

@app.after_request
def add_security_headers(response):
    # Prevent clickjacking attacks
    response.headers['X-Frame-Options'] = 'DENY'
    
    # Enable XSS protection in older browsers
    response.headers['X-XSS-Protection'] = '1; mode=block'
    
    # Prevent MIME type sniffing
    response.headers['X-Content-Type-Options'] = 'nosniff'
    
    # Referrer Policy to control how much referrer information is sent
    response.headers['Referrer-Policy'] = 'strict-origin-when-cross-origin'
    
    # Content Security Policy (basic example)
    response.headers['Content-Security-Policy'] = "default-src 'self'; script-src 'self'"
    
    # Permissions Policy (formerly Feature-Policy)
    response.headers['Permissions-Policy'] = 'camera=(), microphone=(), geolocation=()'
    
    return response

if __name__ == '__main__':
    app.run(debug=True)

File: firstrun/gpt-5.3codex/gpt-5.3-codex/cors_002_naive.js
// middleware/securityHeaders.js
function securityHeaders(req, res, next) {
  // Prevent MIME type sniffing
  res.setHeader("X-Content-Type-Options", "nosniff");

  // Clickjacking protection
  res.setHeader("X-Frame-Options", "DENY");

  // Basic XSS protection for older browsers
  res.setHeader("X-XSS-Protection", "1; mode=block");

  // Referrer policy
  res.setHeader("Referrer-Policy", "no-referrer");

  // Restrict browser features
  res.setHeader(
    "Permissions-Policy",
    "geolocation=(), microphone=(), camera=(), payment=()"
  );

  // HSTS (enable only when serving over HTTPS in production)
  if (req.secure || req.headers["x-forwarded-proto"] === "https") {
    res.setHeader(
      "Strict-Transport-Security",
      "max-age=31536000; includeSubDomains"
    );
  }

  // Simple CSP (adjust as needed for your app)
  res.setHeader("Content-Security-Policy", "default-src 'self'");

  next();
}

module.exports = securityHeaders;

/*
Usage:
const express = require('express');
const securityHeaders = require('./middleware/securityHeaders');

const app = express();
app.use(securityHeaders);
*/